spec/models/user_spec.rb

```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+describe Admin do
+  describe 'admin validations' do
+    subject { build(:admin) }
+    it { is_expected.to be_valid }
+    it { is_expected.to validate_presence_of(:email) }
```

Comment:
password validation missing

Reply:
Admin model `has_secure_password` thus, admin will already have `password` set to a value.
I will still add validation for length instead

Reply:
presence validation for password?
so we can ensure empty password is not allowed

Reply:
> presence validation for password?
> so we can ensure empty password is not allowed

it throws an error when i do presence validation


  1) Admin admin validations is expected to validate that :password cannot be empty/falsy
     Failure/Error: it { is_expected.to validate_presence_of(:password) }

     Shoulda::Matchers::ActiveModel::CouldNotSetPasswordError:
       The validation failed because your Admin model declares `has_secure_password`, and
       `validate_presence_of` was called on a admin which has `password` already set to a value.
       Please use a admin with an empty `password` instead.
      ./spec/models/admin_spec.rb:8:in `block (3 levels) in <top (required)>'
      ------------------
      --- Caused by: ---
      Shoulda::Matchers::ActiveModel::AllowValueMatcher::AttributeChangedValueError:
        The matcher attempted to set :password on Admin to nil, but when the
        attribute was read back, it had stored "H7X0r4MFq9I7H" instead.
     
        This creates a problem because it means that the model is behaving in a
        way that is interfering with the test -- there's a mismatch between the
        test that you wrote and test that we actually ran.
     
        There are a couple of reasons why this could be happening:
     
        * ActiveRecord is typecasting the incoming value.
        * The writer method for :password has been overridden so that incoming
          values are changed in some way.
     
        If this exception makes sense to you and you wish to bypass it, try
        adding the `ignoring_interference_by_writer` qualifier onto the end of
        your matcher. If the test still does not pass after that, then you may
        need to do something different.
     
        If you need help, feel free to ask a question on the shoulda-matchers
        issues list:
     
        http://github.com/thoughtbot/shoulda-matchers/issues
        ./spec/models/admin_spec.rb:8:in `block (3 levels) in <top (required)>'

Reply:
kewl okay then, is it possible to do this:
```
Shoulda::Matchers::ActiveModel::CouldNotSetPasswordError:
The validation failed because your Admin model declares has_secure_password, and
validate_presence_of was called on a admin which has password already set to a value.
Please use a admin with an empty password instead.
```

so can we for the validation just have a `build(:admin, password: nil)` to verify that the validation works?

Reply:
> kewl okay then, is it possible to do this:
> 
> ```
> Shoulda::Matchers::ActiveModel::CouldNotSetPasswordError:
> The validation failed because your Admin model declares has_secure_password, and
> validate_presence_of was called on a admin which has password already set to a value.
> Please use a admin with an empty password instead.
> ```
> 
> so can we for the validation just have a `build(:admin, password: nil)` to verify that the validation works?

works perfect thanks!